gpt_index/indices/knowledge_graph/base.py

```diff
@@ -102,13 +102,10 @@ def _parse_triplet_response(response: str) -> List[Tuple[str, str, str]]:
 
     def _build_index_from_documents(self, documents: Sequence[BaseDocument]) -> KG:
         """Build the index from documents."""
-        text_splitter = self._prompt_helper.get_text_splitter_given_prompt(
```

Comment:
so for all _build_index_from_documents calls, we are now never using the text_splitter returned by the prompt helper? 

imo i liked the logic of 
```
text_splitter = (
      self._text_splitter
      or self._prompt_helper.get_text_splitter_given_prompt(prompt_with_schema, 1)
  )
```
think we should keep for index construction as well. otherwise will alter behavior of existing indices

Reply:
Ah ok, I thought one of the goals of this was to move away from constructing indices that depend on a prompt? For example, in case the user wants to change the prompt or try different ones, the index would have to be regenerated using a different text-splitter each time.

Reply:
Yeah that's something I'd want to think about; at the end of the day the chunk size determined during index construction is an approximation anyways since the prompt can be customized during query-time (and it's actually ok for chunk sizes to be too big/small etc. we handle that under the hood with our response builder). So you shouldn't need to regenerate the index even if you try out diff query prompts.

I figured for now though it's better to keep the existing default behavior if text_splitter isn't specified. Wdyt?

Reply:
That's fine for now. In that case, this PR will be primarily about exposing text_splitter to the user. Some related then, should we expose a `text_splitter_class` argument on `PromptHelper`, or just wait to refactor that altogether?

Reply:
i think it would make sense to expose a text_splitter_class to PromptHelper but lower pri imo; the main pain point this PR solves is to more easily text splitting without needing to deal with the current prompthelper abstraction